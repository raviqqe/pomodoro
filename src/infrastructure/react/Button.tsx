import { FC, HTMLAttributes, HTMLProps, ReactNode } from "react";
import styled from "styled-components";
import { boxShadow } from "./style";

interface IProps
  extends HTMLProps<HTMLButtonElement>,
    HTMLAttributes<HTMLButtonElement> {
  secondary?: boolean;
  children?: ReactNode;
}

export const Button: FC<IProps> = styled.button<{ secondary?: boolean }>`
  ${boxShadow};
  background: ${({ secondary }) => (secondary ? "darkgrey" : "salmon")}
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: inherit;
  color: white;
  border: none;
  cursor: pointer;
  flex-shrink: 0;
` as FC<any>;
