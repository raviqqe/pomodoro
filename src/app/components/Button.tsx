import { styled } from "@linaria/react";
import { grey, red, white } from "./style/colors.js";
import { boxShadow } from "./style.js";

export const Button = styled.button<{ secondary?: boolean }>`
  ${boxShadow};
  background: ${({ secondary }) => (secondary ? grey : red)};
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: inherit;
  color: ${white};
  border: none;
  cursor: pointer;
  flex-shrink: 0;
`;
