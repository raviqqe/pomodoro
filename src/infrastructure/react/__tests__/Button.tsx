import { create } from "react-test-renderer";
import React from "react";
import { Button } from "../Button";

it("renders", () => {
  expect(create(<Button>foo</Button>).toJSON()).toMatchSnapshot();
});

it("renders as a secondary button", () => {
  expect(
    create(<Button secondary={true}>foo</Button>).toJSON()
  ).toMatchSnapshot();
});
