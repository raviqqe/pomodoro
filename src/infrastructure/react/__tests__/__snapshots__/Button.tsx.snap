// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  className="sc-bdVaJa btZbqx"
>
  foo
</button>
`;

exports[`renders as a secondary button 1`] = `
<button
  className="sc-bdVaJa fbvySC"
>
  foo
</button>
`;
